extensions/OrangeFactor/web/js/orange_factor.js

```diff
@@ -53,39 +49,26 @@ $(function() {
         $('#orange-count').text(count + ' failures on trunk in the past week');
     }
 
-    function dateString(date) {
-        function norm(part) {
-            return JSON.stringify(part).length == 2 ? part : '0' + part;
-        }
-        return date.getFullYear()
-            + "-" + norm(date.getMonth() + 1)
-            + "-" + norm(date.getDate());
-    }
-
-    function getCurrentDateMs() {
-        var d = new Date;
-        return d.getTime();
-    };
-
     function orangify() {
-        $('#orange-count')
-            .text('Loading...')
-            .show();
-        var bugId = document.forms['changeform'].id.value;
-        var request = {
+        let $orangeCount = $('#orange-count');
+        let queryParams = $.param({
+            bug: $orangeCount.data('bug-id'),
+            startday: $orangeCount.data('date-start'),
+            endday: $orangeCount.data('date-end'),
+            tree: 'trunk'
+        });
+        let request = {
             dataType: "json",
-            xhrFields: {
-                withCredentials: true
```

Comment:
No need to send creds anymore since the new API allows cross origin requests from everywhere.